commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderItemClayTable.java

```diff
@@ -0,0 +1,576 @@
+
+		BaseModelSearchResult<CommerceOrderItem> baseModelSearchResult = null;
+
+		long commerceOrderId = ParamUtil.getLong(
+			httpServletRequest, "commerceOrderId");
+
+		CommerceOrder commerceOrder = _commerceOrderService.getCommerceOrder(
+			commerceOrderId);
+
+		int start = QueryUtil.ALL_POS;
+		int end = QueryUtil.ALL_POS;
+
+		if (pagination != null) {
+			start = pagination.getStartPosition();
+			end = pagination.getEndPosition();
+		}
+
+		_setSortPreferences(httpServletRequest);
+
+		Sort sort = SortFactoryUtil.getSort(
+			CommerceOrderItem.class, _orderByCol, _orderByType);
+
+		OrderItemFilterImpl orderItemFilter = (OrderItemFilterImpl)filter;
+
+		if (orderItemFilter.isAdvancedSearch()) {
+			baseModelSearchResult = _commerceOrderItemService.search(
+				commerceOrder.getCommerceOrderId(), orderItemFilter.getSku(),
+				orderItemFilter.getName(), orderItemFilter.isAndOperator(),
+				start, end, sort);
+		}
+		else {
+			baseModelSearchResult = _commerceOrderItemService.search(
+				commerceOrder.getCommerceOrderId(),
+				orderItemFilter.getKeywords(), start, end, sort);
+		}
+
+		return baseModelSearchResult;
+	}
+
+	private PortletURL _getOrderItemDeleteURL(
+			long commerceOrderItemId, HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		PortletURL portletURL = PortletProviderUtil.getPortletURL(
+			httpServletRequest, CommerceOrder.class.getName(),
+			PortletProvider.Action.MANAGE);
+
+		portletURL.setParameter(
+			ActionRequest.ACTION_NAME, "editCommerceOrderItem");
+		portletURL.setParameter(Constants.CMD, Constants.DELETE);
+		portletURL.setParameter(
+			"redirect", _portal.getCurrentURL(httpServletRequest));
+		portletURL.setParameter(
+			"commerceOrderItemId", String.valueOf(commerceOrderItemId));
+
+		return portletURL;
+	}
+
+	private String _getOrderItemPanelURL(
+			long commerceOrderItemId, HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		PortletURL portletURL = PortletProviderUtil.getPortletURL(
+			httpServletRequest, CommerceOrder.class.getName(),
+			PortletProvider.Action.MANAGE);
+
+		portletURL.setParameter(
+			"mvcRenderCommandName", "editCommerceOrderItem");
+		portletURL.setParameter(
+			"redirect", _portal.getCurrentURL(httpServletRequest));
+
+		long commerceOrderId = ParamUtil.getLong(
+			httpServletRequest, "commerceOrderId");
+
+		portletURL.setParameter(
+			"commerceOrderId", String.valueOf(commerceOrderId));
+
+		portletURL.setParameter(
+			"commerceOrderItemId", String.valueOf(commerceOrderItemId));
+
+		try {
+			portletURL.setWindowState(LiferayWindowState.POP_UP);
+		}
+		catch (WindowStateException wse) {
+			_log.error(wse, wse);
+		}
+
+		return portletURL.toString();
+	}
+
+	private String _getPeriodKey(
+		String period, boolean plural, HttpServletRequest httpServletRequest) {
+
+		if (plural) {
+			return LanguageUtil.get(
+				httpServletRequest,
+				StringUtil.toLowerCase(period) + CharPool.LOWER_CASE_S);
+		}
+
+		return period;
+	}
```

Comment:
Same comments on this as https://github.com/liferay/com-liferay-commerce/pull/1504#pullrequestreview-300271386.